src/charm.py

```diff
@@ -164,13 +188,25 @@ def install_sdc(self):
             logger.error("The package required for SDC installation is missing")
             return
 
+        # Store the name of the SDC package for later use.
+        result = subprocess.run(
+            ["dpkg", "--info", str(sdc_package_file)],
+            capture_output=True,
+            text=True,
+        )
+        mo = re.search(r"^\s*Package:\s+(.+?)$", result.stdout, re.MULTILINE)
+        if mo:
+            self._stored.sdc_package_name = mo.group(1)
+        else:
+            logger.warning("Couldn't determine package name from %s", sdc_package_file)
+
         # Get the MDM IP from config file
         sdc_mdm_ips = config["powerflex-sdc-mdm-ips"]
         # Install the SDC package
-        install_cmd = f"sudo MDM_IP={sdc_mdm_ips} dpkg -i {sdc_package_file}"
+        install_cmd = ["sudo", "env", f"MDM_IP={sdc_mdm_ips}", "dpkg", "-i", str(sdc_package_file)]
```

Comment:
Out of interest, why the addition of `env`?

Reply:
Yeah, it was accidental. I should also mention this change in the PR description since it's not related to the remove. While I was doing the remove command I noticed the `.split()` in the command and felt that ought to be improved since there's unverified user input in there.

I wrote the new `install_cmd` by hand rather than editing the existing line, and either typed in `env` from habit or possibly tabbed in a (not ideal) suggestion. Will fix.